src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -4716,9 +4723,10 @@ class B : A
             // PROTOTYPE(NullableReferenceTypes): Should report return type mismatch
             // for M1 and M2 (see https://github.com/dotnet/roslyn/issues/28684).
```

Comment:
>https://github.com/dotnet/roslyn/issues/28684 [](start = 34, length = 45)

It looks like you've fixed #28684. Is the issue with `M2<S>` a case where the meaning of `S?` is unclear since `S` is unconstrained? If so, that should be resolved when we report an error for use of `T?` for unconstrained `T`.